sklearn/cluster/_k_means_minibatch.pyx

```diff
@@ -0,0 +1,228 @@
+# cython: profile=True, boundscheck=False, wraparound=False, cdivision=True
```

Comment:
do we need it ? No.
do we want it ? it's interesting to have it, it makes it possible to profile KMeans beyond pure python functions with standard python profiler like cprofile. I agree that it can add some overhead, however I benchmarked with and without at some point and never saw any difference. Moreover, the directive was already there prior to this PR. I think we can postpone the discussion of keeping it to another PR (it might actually be interesting to remove it locally and add a global option to compile sklearn with or without this directive)